src/ping.js

```diff
@@ -11,17 +11,22 @@ var Ping = function() {
  * Pings source and triggers a callback when completed.
  * @param source Source of the website or server.
  * @param callback Callback function to trigger when completed.
+ * @param timeout Optional number of milliseconds to wait before aborting.
  */
-Ping.prototype.ping = function(source, callback) {
+Ping.prototype.ping = function(source, callback, timeout) {
     this.img = new Image();
+    timeout = timeout || 0;
+    var timer;
 
     var start = new Date();
     this.img.onload = this.img.onerror = pingCheck;
+    if (timeout) timer = setTimeout(pingCheck,timeout);
```

Comment:
Looks good! Can you add the space before `timeout` and we'll be good! Thank you!